extension/src/fileSystem/index.ts

```diff
@@ -214,21 +214,38 @@ const loadYamlAsDoc = (
   }
 }
 
+const getPlotYamlObj = (cwd: string, plot: PlotConfigData) => {
+  const { x, y, template } = plot
+  const usesSingleFile = x.file === y.file
```

Comment:
[Q] Do we need this split? Can we make the creation of `plots` entries a bit more opinionated from the wizard and reduce complexity? Otherwise, when we come to add in custom titles we will have to complicate the differentiation further.

From our demo `dvc.yaml`:

```
  - Loss:
      x: step
      y:
        training/plots/metrics/train/loss.tsv: loss
        training/plots/metrics/test/loss.tsv: loss
      y_label: loss
  - Confusion matrix:
      template: confusion
      x: actual
      y:
        training/plots/sklearn/confusion_matrix.json: predicted
  - hist.csv:
      x: preds
      y: digit
      template: bar_horizontal
      title: Histogram of Predictions
```

Maybe we want to stick to the first two entry types.

Reply:
See also https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#available-configuration-fields:

![image](https://github.com/iterative/vscode-dvc/assets/37993418/6ed3ce0d-4272-485a-a144-d587926c4d34)

Reply:
Updated the two plot types to be more similar:

```yaml
plots:
  # two files
  - scatter_plot:
      template: scatter
      x:
        props.json: acc
      y:
        values.json: prob
  # single files
  - simple_plot:
      template: simple
      x: actual
      y:
        probs.json: prob
```